src/repl/scala/tools/nsc/interpreter/session/FileBackedHistory.scala

```diff
@@ -80,5 +80,5 @@ object FileBackedHistory {
   import Properties.userHome
 
   def defaultFileName = ".scala_history"
-  def defaultFile: File = File(Path(userHome) / defaultFileName)
+  def defaultFile: File = File( Path( Properties.propOrElse( "scala.repl.histfile", userHome +"/"+ defaultFileName) ) )
```

Comment:
I think slash is OK internally on Windows.

Less cleverly, `Option(propOrNull) map (File(_)) getOrElse File(userHome, default)`. Or whatever.